tests/test-ucx-w-dask-workers.py

```diff
@@ -0,0 +1,199 @@
+
+                        res_x = res_x.persist(workers=[w1.worker_address])
+                        res_y = res_y.persist(workers=[w2.worker_address])
+                        
+                        res = (res_x + res_y).persist()
+                        out = await c.compute(res.head(compute=False))
+                        print(out)
+
+async def h(left_n_rows):
+    async with Scheduler(protocol=protocol, interface=interface,
+    dashboard_address=':8789') as s:
+        async with Nanny(s.address, protocol=protocol, nthreads=1,
+                memory_limit='32GB', interface=interface,
+                env=w_env,
+                ) as w1:
+            async with Nanny(s.address, protocol=protocol,memory_limit='32gb',
+                    env=w_env_2, interface=interface,
+                    nthreads=1) as w2:
+                async with Client(s.address, asynchronous=True) as c:
+                    with log_errors():
+                        import dask.array as da
+                        import dask_cudf
+                        import numpy as np
+                        import cupy
+                        import cudf
+
+                        def set_nb_context(x=None):
+                            import numba.cuda
+                            try:
+                                numba.cuda.current_context()
+                            except Exception:
+                                print("FAILED EXCEPTION!")
+                                
+                        out = await c.run(
+                            set_nb_context,
+                            workers=[w1.worker_address, w2.worker_address]
+                        )
+                        
+                        # left = dd.concat([
+                        #     da.random.random(left_n_rows).to_dask_dataframe(columns='x'),
+                        #     da.random.randint(0, 10, size=left_n_rows).to_dask_dataframe(columns='id'),
+                        # ], axis=1)
+
+                        # left = await left.map_partitions(cudf.from_pandas).persist(workers=[w1.worker_address])
+                        # print(f"Left npartitions: {left.npartitions}")
+
+                        # out = await left.repartition(npartitions=10).persist()
+
+
+                        future = c.submit(dataframe, left_n_rows)
+
+                        npartitions = 10
+                        results = []
+                        for i in range(npartitions):
+                            start = i * (left_n_rows // npartitions)
+                            end = (i + 1) * (left_n_rows // npartitions)
+                            worker = (w1.worker_address, w2.worker_address)[i % 2]
+                            results.append(c.submit(lambda x: x.iloc[start:end], future, workers=[worker]))
+
+                        del future
+                        for result in results:
+                            await result
```

Comment:
@mrocklin @quasiben 

The latter part of this test and the region commented above it should both do roughly the same thing - correct?

Only the commented region produces a hang.